comp/otelcol/collector-contrib/impl/go.mod

```diff
@@ -765,6 +767,12 @@ replace github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/metricscli
 
 replace github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/statsprocessor => ../../otlp/components/statsprocessor
 
+replace github.com/DataDog/datadog-agent/comp/trace/compression/def => ../../../trace/compression/def
+
+replace github.com/DataDog/datadog-agent/comp/trace/compression/impl-gzip => ../../../trace/compression/impl-gzip
+
+replace github.com/DataDog/datadog-agent/comp/trace/compression/impl-zstd => ../../../trace/compression/impl-zstd
```

Comment:
We can remove this replace as we no longer use zstd implementation in OTel

Reply:
all OTel modules that depend on `/pkg/trace/` end up requiring all these implementations, at least in their go.mod, since `pkg/trace/writer` is still directly using them for testing purposes.

We could decouple the component from the trace-writer tests and somewhat simplify these dependencies, but that would be a separate PR.